packages/z2s/src/compiler.ts

```diff
@@ -596,20 +596,29 @@ export class Compiler {
         this.#nameMapper.columnName(table, column)
       ];
     const serverType = serverColumnSchema.type;
-    if (
-      !serverColumnSchema.isEnum &&
-      (serverType === 'date' ||
+    if (!serverColumnSchema.isEnum) {
+      if (
+        serverType === 'date' ||
         serverType === 'timestamp' ||
+        serverType === 'timestamp without time zone'
+      ) {
+        return sql`EXTRACT(EPOCH FROM ${sql.ident(
+          table,
+        )}.${this.#mapColumnNoAlias(
+          table,
+          column,
+        )}::timestamp AT TIME ZONE 'UTC') * 1000 as ${sql.ident(column)}`;
+      } else if (
         serverType === 'timestamptz' ||
-        serverType === 'timestamp with time zone' ||
-        serverType === 'timestamp without time zone')
-    ) {
-      return sql`EXTRACT(EPOCH FROM ${sql.ident(
-        table,
-      )}.${this.#mapColumnNoAlias(
-        table,
-        column,
-      )}::timestamp AT TIME ZONE 'UTC') * 1000 as ${sql.ident(column)}`;
+        serverType === 'timestamp with time zone'
+      ) {
+        return sql`EXTRACT(EPOCH FROM ${sql.ident(
+          table,
+        )}.${this.#mapColumnNoAlias(
+          table,
+          column,
+        )}::timestamp) * 1000 as ${sql.ident(column)}`;
```

Comment:
I think this should just be:

```suggestion
        return sql`EXTRACT(EPOCH FROM ${sql.ident(
          table,
        )}.${this.#mapColumnNoAlias(
          table,
          column,
        )}) * 1000 as ${sql.ident(column)}`;
```

as that round-trips correctly (1744392199 in and 1744392199 out):

```
user@127.0.0.1:postgres> create table foo (a timestamp with time zone);
CREATE TABLE
Time: 0.028s
user@127.0.0.1:postgres> INSERT INTO foo VALUES (to_timestamp(1744392199));
INSERT 0 1
Time: 0.004s
user@127.0.0.1:postgres> select * from foo;
+------------------------+
| a                      |
|------------------------|
| 2025-04-11 12:23:19-05 |
+------------------------+
SELECT 1
Time: 0.006s
user@127.0.0.1:postgres> SELECT EXTRACT(EPOCH FROM "a") FROM foo;
+-------------------+
| extract           |
|-------------------|
| 1744392199.000000 |
+-------------------+
SELECT 1
Time: 0.008s
user@127.0.0.1:postgres>
```

Reply:
this is just because the `::timestamp` cast is redundant?

Reply:
and I guess `EPOCH FROM` with a `timestamp` just treats it as UTC

Reply:
sorry read it again. This was a `timestamptz`, and the cast was stripping the tz, and then we were treating it as local time and converting that to utc